src/main/scala/net/snowflake/spark/snowflake/io/StageWriter.scala

```diff
@@ -281,16 +281,19 @@ private[io] object StageWriter {
     val writeTableState = new WriteTableState(conn)
 
     try {
+      val tableExists = if (params.checkTableExistenceInCurrentSchemaOnly){
+        DefaultJDBCWrapper.tableExistsInCurrentSchema(params, tableName)
+      } else {
+        DefaultJDBCWrapper.tableExists(conn, tableName)
+      }
       // Drop table only if necessary.
-      if (saveMode == SaveMode.Overwrite &&
-        DefaultJDBCWrapper.tableExists(conn, tableName) &&
-        !params.truncateTable )
+      if (saveMode == SaveMode.Overwrite && tableExists && !params.truncateTable)
```

Comment:
Does this code handle all of the different `SaveMode`s properly? Consider also `Ignore` and `ErrorIfExists`.

Reply:
Excellent catch! This change only affects `OverWrite` and `Append`. `Ignore` and `ErrorIfExists` are handled in `DefaultSource`.  I will add tests and update patch for them.